### DISCUSSION TITLE
Link WalletConnet/Metamask with Openlogin

### DISCUSSION THREAD

**Discussion by culda** [Web3Auth/Web3Auth#909] [Help]
I want to start a discussion about what I'm trying to achieve with Web3Auth.

Users sign up with their metamask (or other wallet) and I save their web3 public key in my DB.
They can then add some social login like Twitter. I will then create a link in my DB between their Twitter id and their web3 public key.

Some problems that I have.
- I've got metamask login set up, but I can't find a way to get the public key. `getUserInfo()` returns empty
- I can't get WalletConnet to work. Is it because v2 is not supported yet and my app is v1 maybe? I can't tell what version my app is from my walletconnect dashboard

**Comment by Saviour1001** [comment]
- When connecting with Metamask or other wallets, you can get the public key using `getAccounts` function depending upon the EVM library you are using (ethers,web3js). Here is an [example](https://github.com/Web3Auth/examples/blob/2151a2897a3f77ce42fff32200fd582e6c07803e/web-modal-sdk/evm/react-evm-web3auth-example/src/web3RPC.ts#L37) depicting the same.  
- `getUserInfo()` only works for social logins, not external wallets. 
- The WalletConnect v2 is not [backwards compatible](https://docs.walletconnect.com/2.0/advanced/migrating-from-v1.0#migration-schedule). Can you provide screenshots of the errors when using WalletConnect?
